### PR TITLE
BayMerge DLC Pack

### PR DESCRIPTION
**Pull request by LordFowl**
Ports totally;
https://github.com/Baystation12/Baystation12/pull/12389/files
https://github.com/Baystation12/Baystation12/pull/14599
Ports partially;
https://github.com/Baystation12/Baystation12/pull/14216 (only blob-related functions)

All the failure messages have been replaced with failure events in industrial drills, that range from probably lethal to certainly lethal and in one case just moderately annoying.

### CODE REVIEW

File: code/modules/blob/blob.dm [hunk 1]
```diff
@@ -4,16 +4,21 @@
...
 	opacity = 0
 	anchored = 1
 	mouse_opacity = 2
+	layer = 4
 
 	var/maxHealth = 30
 	var/health
+	var/regen_rate = 5
 	var/brute_resist = 4
+	var/laser_resist = 5	// Special resist for laser based weapons - Emitters or handheld energy weaponry. Damage is divided by this and THEN by fire_resist.
```
Comment (NanakoAC) [inline code/modules/blob/blob.dm]: i approve of laser resistance but this seems like a crazy amount. This amounts to 80% damage reduction from lasers. 
They ALSO have 75% reduction from brute damage, so what are you actually supposed to hit them with?
At this point you might as well just increase their health

I agree blobs need a buff, but i don't think making them this much more powerful is a good idea.

Whats the design goal here? How do you want crew to fight blobs?

Comment (LordFowl) [inline code/modules/blob/blob.dm]: Flamethrowers and welding torches and everything else that causes burn damage.

Comment (NanakoAC) [inline code/modules/blob/blob.dm]: flamethrowers are broken and worthless. They need a lot of fixing before they can be considered for use as weapons

Welding torches are tools, not weapons. They're not bad, but theres something grossly wrong if a hand welder isn't any more efficient than a giant laser wired to the station's power grid.

Have you tested these changes, and attempted to fight a blob with these methods?

I agree with the direction, lasers are overused and could do with some reduced effectiveness. But these values are far too much

Comment (NanakoAC) [inline code/modules/blob/blob.dm]: Did a bit of quick testing to confirm, i'm right about flamethrowers. Baymerge has not fixed them, they are still broken and worthless.

The fire from a flamethrower does nothing to a blob. Literally nothing, zero damage. The only thing that does damage is bludgeoning a blob with the thing, and thats still barely enough to kill a piece of a blob before it eats you

**Comment by NanakoAC** [comment]
my recommended values for the laser resistance would be 1.5 for normal blob parts and 2 for the core. at a base they increase the survival times by 50/100% respectively, but when combined with constant regen, they will add a lot more to total time to kill than those values would imply

**Comment by NanakoAC** [comment]
And now looking closer, i see you've also quintupled their regen rate.
Blob regen was not an insignificant mechanic before, the things constantly regrow new parts

**Comment by LordFowl** [comment]
Dealing with blobs currently is a non-issue. They are laughable compared to other major events, especially bluespace bears. People die to bluespace bears, to migratory carp, to virus epidemics. People do not die to blobs. These are major buffs yes, but I believe they are warranted considering the blob's role.

**Comment by NanakoAC** [comment]
I agree with you on some of those points. Blobs don't kill much, and they should be more lethal.

However i strongly disagree that your proposed buffs will accomplish this goal, or even move towards it in the slightest.

Ive never seen anyone get eaten by a blob. But i have seen several blob related deaths. The overwhelming amount of these were due to hull breaches it caused when spawning, and a few were due to accidents with the emitter. Generally, overzealous security walking infront of an emitter

If you want to make them more lethal, you need to increase their offensive power. Stuns, bursts of rapid growth, tentacles to grab and pull people in, whatever.  There's a lot to think about.
Making them tougher does not make them better at killing people. It just makes them boring to kill, at best. And because of the insane severity of your buffs, there's a real risk of the blobs being deemed unkillable, their location cordoned off, and just ignored for the rest of the round. Likely not long before an emergency shuttle. Ive seen that happen sometimes, 

If it were bears, then things would be different. The bluespace bears are a mobile, offensive threat. Increasing their health directly increases their killing power because they're adept at closing distance and striking. How much damage they deal is a product of how long they live.

Blobs are a stationary threat. They eat and breach a bit of the station and then just sit there looking icky. Right now however, they are very, very difficult for security to deal with, without calling engineers for help. Certainly emitters do kill them easily. Even too easily, that i agree with. But it does encourage interdepartmental cooperation.

If you nerf emitters into worthlessness, as you're planning. What does that leave? Welders are the only viable thing you've mentioned.  Now the blobs are melee fighters. They burn people in melee. An engineer emittering them from afar is generally acceptable because of the minimal risk. But asking engineers to use their welding tools as weapons and charge into the fray is imo not acceptable., that's playing hero, and violates the rule of avoiding pain. Any sane engineer, if ordered to cut up a blob with welding tools, would tell their CO to shove it up their ass, they're not being paid enough for that.

Their 5 brute reduction ensures security can't beat the blobs to death with batons either. Does that apply to ballistic weapons ? I don't know, thats worth looking into. But that possibility aside, it seems like blobs are designed to be emittered. Emitters are their counter, as EMPs to robots. or spears to carp

In short:
More durability doesn't make blobs better at killing
Your changes don't fulfil the design goal of making them more dangerous.
Nerfing emitters will marginalise interdepartmental cooperation and/or encourage bad RP from engineers

If you want them to be more than trivial to deal with, these changes are fine in principle but the values are way too high. 1.5/2 is best imo.  If you want them to be an actual threat, you need to look in a different direction. Even making them literally invincible wouldn't help them kill people.

**Comment by LordFowl** [comment]
After some additional testing I may reduce the values slightly, the base one from a 5 to perhaps a 4 or possibly a 3. With 5 it is impossible to use an emitter alone, and with 3 an emitter will eventually take care of it all, but it will take longer, so I think 4 is the sweet spot that requires concentrated laser fire, or welders which still take care of the situation handily.

**Comment by skull132** [comment]
Then run that commit. Currently it's the only thing holding the PR back.

**Comment by skull132** [comment]
Still waiting, @LordFowl

**Comment by LordFowl** [comment]
Quite a few updates, probably poorly balanced and in need of review. Got a bit carried away on this one, lads.

File: code/modules/blob/blob.dm [hunk 2]
```diff
@@ -50,74 +114,59 @@
...
 		if(L.stat == DEAD)
 			continue
-		L.visible_message("<span class='danger'>The blob attacks \the [L]!</span>", "<span class='danger'>The blob attacks you!</span>")
-		playsound(loc, 'sound/effects/attackblob.ogg', 50, 1)
-		L.take_organ_damage(rand(30, 40))
+		if(prob(30))
+			L.visible_message("<span class='danger'>The blob sucks \the [L] into itself!</span>", "<span class='danger'>The blob sucks you in!</span>")
+			playsound(loc, 'sound/effects/attackblob.ogg', 50, 1)
+			L.take_organ_damage(rand(5, 10))
+			L.loc = src.loc
```
Comment (skull132) [inline code/modules/blob/blob.dm]: `L.forceMove(src.loc)`

File: code/modules/blob/blob.dm [hunk 3]
```diff
@@ -145,6 +195,11 @@
 			if(istype(W, /obj/item/weapon/weldingtool))
 				playsound(loc, 'sound/items/Welder.ogg', 100, 1)
 		if("brute")
+			if(prob(30))
+				visible_message("<span class='danger'>\The [W] gets caught in the gelatinous folds of \the [src]</span>")
+				user.drop_from_inventory(W)
+				W.loc = src.loc
```
Comment (skull132) [inline code/modules/blob/blob.dm]: `W.forceMove(src.loc)`

File: code/modules/mining/drilling/drill.dm [hunk 4]
```diff
@@ -141,14 +141,22 @@
...
 			return
 		if(default_part_replacement(user, O))
 			return
-	if(!panel_open || active) return ..()
+	if(active) return ..()
+
+	if(istype(O, /obj/item/weapon/crowbar))
+		if (panel_open && cell)
+			user << "You wrench out \the [cell]."
+			cell.loc = get_turf(user)
```
Comment (skull132) [inline code/modules/mining/drilling/drill.dm]: `cell.forceMove(get_turf(user))`

File: code/modules/mining/drilling/drill.dm [hunk 5]
```diff
@@ -189,8 +191,44 @@
...
+					s.start()
+					sleep(10)
+					if(panel_open)
+						if(prob(70))
+							visible_message("<span class='danger'>\The [src]'s cell shorts out!</span>")
+							cell.use(cell.charge)
+						else
+							visible_message("<span class='danger'>\The [src]'s cell detonates!</span>")
+							explosion(src.loc, -1, -1, 2, 1)
+							qdel(cell)
```
Comment (skull132) [inline code/modules/mining/drilling/drill.dm]: Make sure to set `cell = null` and `component_parts -= cell` before qdel? Unless this is meant to destroy the entire thing.

File: code/modules/mining/drilling/drill.dm [hunk 6]
```diff
@@ -283,7 +322,9 @@
 			O.loc = B
 		usr << "<span class='notice'>You unload the drill's storage cache into the ore box.</span>"
 	else
-		usr << "<span class='notice'>You must move an ore box up to the drill before you can unload it.</span>"
+		for(var/obj/item/weapon/ore/O in contents)
+			O.loc = src.loc
```
Comment (skull132) [inline code/modules/mining/drilling/drill.dm]: `O.forceMove(src.loc)`

**Comment by skull132** [comment]
@NanakoAC needs to update her review.

File: code/modules/blob/blob.dm [hunk 7]
```diff
@@ -4,23 +4,87 @@
...
 	update_icon()
 	return ..(loc)
 
+/obj/effect/blob/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+/obj/effect/blob/process()
+	if(!parent_core)
+		src.take_damage(5)
```
Comment (NanakoAC) [inline code/modules/blob/blob.dm]: tiny thing, i'd say make them die off slower without the parent. At this ray theres not much difference to just qdeling them, no pleasure in watching it die

File: code/modules/blob/blob.dm [hunk 8]
```diff
@@ -4,23 +4,87 @@
...
+			L.visible_message("<span class='danger'>The blob absorbs \the [L]!</span>", "<span class='danger'>The blob absorbs you!</span>")
+			playsound(loc, 'sound/effects/attackblob.ogg', 50, 1)
+			L.take_organ_damage(rand(5, 10))
+			if(health < maxHealth)
+				health += rand(1,10)
+				if(health > maxHealth)
+					health = maxHealth
+			hangry += 16
+
+		for(var/mob/living/L in range(src,"3x3"))
```
Comment (NanakoAC) [inline code/modules/blob/blob.dm]: what does this 3x3 thing do?
Is that a 3x3 square centred on the blob? if so thats no different from a range of 1

File: code/modules/blob/blob.dm (hunk 7 above)
Comment (LordFowl) [inline code/modules/blob/blob.dm]: As process does not actually occur every tick, this actually means it takes quite a while for the blob to die. Perhaps 30 seconds to a minute for basic cores, and over a minute for more complex cores.

File: code/modules/events/false_alarm.dm [hunk 9]
```diff
@@ -8,24 +8,27 @@
...
 		EM = null
 
 /datum/event/false_alarm/announce()
 	var/datum/event_container/EC
-	if (prob(60))
+	EC = event_manager.event_containers[EVENT_LEVEL_MODERATE]
+	/*if (prob(60))
 		EC = event_manager.event_containers[EVENT_LEVEL_MODERATE]
 	else
-		EC = event_manager.event_containers[EVENT_LEVEL_MAJOR]
```
Comment (NanakoAC) [inline code/modules/events/false_alarm.dm]: Why have you done this? Don't remove major events from the fake list.

There are measures to solve problems with any specific event, including excluding it specifically with the no-fake flag

File: code/modules/blob/blob.dm (hunk 8 above)
Comment (LordFowl) [inline code/modules/blob/blob.dm]: If so, one is just as good as the other. I have found more success with a defined 3x3 area, however.

File: code/modules/events/false_alarm.dm (hunk 9 above)
Comment (LordFowl) [inline code/modules/events/false_alarm.dm]: Accidentally left in when debugging, hence the comments and not the deletion.

File: code/modules/mining/drilling/drill.dm [hunk 10]
```diff
@@ -189,8 +191,46 @@
...
+					sleep(10)
+					if(panel_open)
+						if(prob(70))
+							visible_message("<span class='danger'>\The [src]'s cell shorts out!</span>")
+							cell.use(cell.charge)
+						else
+							visible_message("<span class='danger'>\The [src]'s cell detonates!</span>")
+							explosion(src.loc, -1, -1, 2, 1)
+							qdel(cell)
+							cell = null
```
Comment (skull132) [inline code/modules/mining/drilling/drill.dm]: Removing null from component_parts is not really useful. I might have been a little incoherent when correcting you last night, as I was 2 seconds from passing out.

```
qdel(cell)
component_parts -= cell
cell = null
```

This is the proper order of operations.
